app/models/RsFormMappings.scala

```diff
@@ -175,13 +175,26 @@ object RsFormMappings {
       so.length match {
         case x => if (x <= 8) true else false
       }
+    case "schemeOrganiserFields.companyRegistrationNum" =>
+      so.length match {
+        case x => if (x <= 8) true else false
```

Comment:
while at it, would you mind simplifying those lines into 'so.length <= 8'? there's no point matching here.
the method deserves to be changed into something way simpler, but that is a quick starting fix.

Reply:
Done the changes, replaced the code with so.length <= 8